app/models/notification.rb

```diff
@@ -36,7 +38,7 @@ class Notification < ApplicationRecord
   after_commit :publish, :on => :create
 
   validate(:on => :create) do
-    if !has_alert? && self.sound.nil? && self.badge.nil?
+    if !has_alert? && self.sound.nil? && self.badge.nil? && !auth_key.environment.voip?
```

Comment:
This validation requires the auth key to be provided but it might not be (which would be a validation error in itself).

```ruby
&& !(auth_key && auth_key.environment.voip?)
```